test/jdk/tools/jar/ValidatorTest.java

```diff
@@ -0,0 +1,334 @@
+                new EntryNameTestCase("./Hello.txt", true),
+                new EntryNameTestCase("../Hello.txt", false),
+                new EntryNameTestCase(".\\Hello.txt", false),
+                new EntryNameTestCase("..\\Hello.txt", false),
+                new EntryNameTestCase("C:\\Hello.txt", false),
+                new EntryNameTestCase("D:/Hello.txt", false),
+                new EntryNameTestCase("foo\\bar.txt", false),
+                new EntryNameTestCase("foo/bar.txt", true),
+                new EntryNameTestCase("foo/../bar.txt", false),
+                new EntryNameTestCase("foo/./bar.txt", true),
+                new EntryNameTestCase("..", false),
+                new EntryNameTestCase(".", false),
+                new EntryNameTestCase("/home/foo.txt", false),
+                new EntryNameTestCase("./home/foo.txt", true),
+                new EntryNameTestCase("../home/foo.txt", false),
+                new EntryNameTestCase("foo/bar/..", false),
+                new EntryNameTestCase("foo/bar/.", true),
+                new EntryNameTestCase("/foo/bar/../../myhome/bin", false),
+                new EntryNameTestCase("/foo/bar/././myhome/bin", false),
+                new EntryNameTestCase("myHome/..valid", true),
+                new EntryNameTestCase("myHome/.valid", true),
+                new EntryNameTestCase("..valid", true),
+                new EntryNameTestCase(".valid", true)
+        );
+    }
+
+    private List<String> createInvalidEntryJar(Path path) throws IOException {
+        System.out.printf("%n%n*****Creating Jar with the invalid entry names*****%n%n");
+        var out = new ByteArrayOutputStream(1024);
+        List<String> invalidEntryNames;
+        try (var zos = new ZipOutputStream(out)) {
+            invalidEntryNames = zipEntryPaths()
+                .filter(testCase -> {
+                        try {
+                            zos.putNextEntry(new ZipEntry(testCase.entryName()));
+                            var content = "Content of " + testCase.entryName();
+                            zos.write(content.getBytes(StandardCharsets.UTF_8));
+                            return !testCase.isValid();
+                        } catch (IOException ioe) {
+                            throw new UncheckedIOException(ioe);
+                        }
+                    })
+                .map(EntryNameTestCase::entryName).toList();
+            zos.flush();
+        } catch (UncheckedIOException uioe) {
+            throw uioe.getCause();
+        }
+        Files.write(path, out.toByteArray());
+        return invalidEntryNames;
+    }
+
+    @Test
+    public void testValidateJar() throws IOException {
+        var zip = Path.of("Regular.jar");
+        writeManifestAsFirstSecondAndFourthEntry(zip, false, false);
+        try {
+            jar("--validate --file " + zip.toString());
+        } catch (IOException e) {
+            fail("Expecting zero exit code");
```

Comment:
I think we should remove the try/catch block and let the IOException propagate. That way, if htis fails for whatever reason, we get the entire exception stacktrace. A comment before the `jar("--validate --file ...)` line can say that we except the `jar()` call to succeed with a zero exit code.